drud/ddev
# Add 'ddev composer' command, closes #1131

## The Problem/Issue/Bug:
#1131 

## How this PR Solves The Problem:
Adds a `ddev composer` command, allowing for seamless use of composer within the context of the container. Specific support has been added for `ddev composer create-project`, installing the project into `/tmp/composer` and moving the resulting project into the app root when complete.

## Manual Testing Instructions:
- Execute composer commands with `ddev composer`
- Create projects using `ddev composer create-project`
- Test `create-project` args and flags: one argument should be interpreted as the package, two should be interpreted as the package and the installation directory, and three should be interpreted as the package, installation directory, and the version
- Test something like `ddev composer create-project --no-dev -v typo3/cms-base-distribution myproj ^8` and ensure it installs successfully to `${APP_ROOT}/myproj`
- Test other composer commands with and without flags, like: `ddev composer require --quiet twitter/bootstrap` and confirm that commands behave as if arguments are passed directly to composer

**Note:** `ddev` will make no effort to parse any flags provided to it, they will instead be passed to composer.

## Automated Testing Overview:
composer_test.go has been added to test `ddev composer` and `ddev composer create-project` commands.

## Related Issue Link(s):
#1131

## Release/Deployment notes:
